app/Community/Components/UserProfileMeta.php

```diff
@@ -0,0 +1,227 @@
+            $openTickets = array_sum(countOpenTicketsByDev($user->User));
+        }
+
+        $ticketsClosedResolved = getNumberOfTicketsClosedForOthers($user->User);
+        $ticketsResolvedForOthers = 0;
+        foreach ($ticketsClosedResolved as $ticketData) {
+            $ticketsResolvedForOthers += $ticketData['ResolvedCount'];
+        }
+
+        return compact(
+            'gameAuthoredAchievementsCount',
+            'openTickets',
+            'ticketsResolvedForOthers',
+            'totalAuthoredCodeNotes',
+            'totalAuthoredLeaderboards',
+        );
+    }
+
+    private function buildSocialStats(User $user): array
+    {
+        $userSetRequestInformation = getUserRequestsInformation($user);
+        $numForumPosts = $user->forumPosts()->count();
+
+        return compact(
+            'numForumPosts',
+            'userSetRequestInformation',
+        );
+    }
+
+    private function calculateAverageFinishedGames(array $userJoinedGamesAndAwards): string
+    {
+        $totalGames = count($userJoinedGamesAndAwards);
+        $finishedGames = 0;
+
+        // Iterate over each game to check if it is finished.
+        foreach ($userJoinedGamesAndAwards as $game) {
+            if (isset($game['HighestAwardKind']) && $game['HighestAwardKind'] !== null) {
+                $finishedGames++;
+            }
+        }
+
+        // Calculate the average percentage of finished games.
+        $averageFinishedGames = 0;
+        if ($totalGames > 0) {
+            $averageFinishedGames = ($finishedGames / $totalGames) * 100;
+        }
+
+        // Format and return the result to 2 decimal places.
+        return number_format($averageFinishedGames, 2, '.', '');
+    }
+
+    private function calculateAveragePointsPerWeek(string $userMemberSince, int $points = 0): int
+    {
+        $memberSince = Carbon::createFromFormat('Y-m-d H:i:s', $userMemberSince);
+        $now = Carbon::now();
+
+        $weeksOfMembership = $memberSince->diffInWeeks($now);
+
+        // Avoid division by zero.
+        if ($weeksOfMembership == 0) {
+            return $points;
+        }
+
+        $averagePointsPerWeek = $points / $weeksOfMembership;
+
+        return (int) round($averagePointsPerWeek);
+    }
+
+    private function calculateRecentPointsEarned(User $user, string $preferredMode = 'hardcore'): array
+    {
+        $thirtyDaysAgo = now()->subDays(30)->startOfDay();
+
+        $dateColumn = $preferredMode === 'hardcore' ? 'unlocked_hardcore_at' : 'unlocked_at';
+
+        $achievements = $user->playerAchievements()
+            ->with('achievement')
+            ->where($dateColumn, '>=', $thirtyDaysAgo)
+            ->get();
+
+        $pointsLast30Days = 0;
+        $pointsLast7Days = 0;
+
+        $now = now();
+
+        foreach ($achievements as $playerAchievement) {
+            $achievementDate = $playerAchievement->{$dateColumn};
+            $daysAgo = $now->diffInDays($achievementDate, false);
```

Comment:
This logic is backwards. If I print "$daysAgo", it's always negative.

The problem is fairly obvious with our test data as every user appears to have been active in the last week.

Reply:
Thanks, I've adjusted this logic in latest and hopefully it's in a better place now.